klondikemarlen/legal-files-integration
# Issue 5: Fix Deployment Setup

Fixes [#5](https://github.com/klondikemarlen/legal-files-integration/issues/5)

# User Story

A developer can run a `docker compose build && docker compose up` and the app boots correctly.
An external user can post from a Drupal form to the public URL of the project running on Digital Ocean.

# Implementation Details

- Drop Rollup build system and replace it with Parcel.
- Fix build issues by switching from dynamic imports in `src/model.js` to manual imports.
- Add a User Acceptance Testing (UAT) environment config.

# Testing Instructions

1. Create a `.env` file in the top-level project directory from the
   `env.template` file.
2. Build the app via `docker compose -f docker-compose.uat.yaml build`.
3. Deploy the build app via `docker compose -f docker-compose.uat.yaml up`
4. Post example data to the `http://64.227.99.157/api/form-b` endpoint.

```bash
curl --request POST http://64.227.99.157/api/form-b \
  --header "Content-Type: application/json" \
  --data '
{
  "name": "linuxize",
  "email": "linuxize@example.com"
}
'
```

5. Check that the returned data is correct. It should look like:

```
{
	"name": "linuxize",
	"email": "linuxize@example.com"
}
```
